tests/test_filter.py

```diff
@@ -227,3 +229,59 @@ def test_crossover(impulse):
     assert f_obj.comment == ("Linkwitz-Riley cross over network of order 2 at "
                              "100, 10000 Hz.")
     x = pfilt.crossover(impulse, 2, [100, 10_000])
+
+
+def test_reconstructing_fractional_octave_bands():
+    """Test the reconstructing fractional octave filter bank."""
+
+    # test filter object
+    f_obj, f = pfilt.reconstructing_fractional_octave_bands(
+        None, sampling_rate=44100)
+    assert isinstance(f_obj, pclass.FilterFIR)
+    assert f_obj.comment == \
+        ("Perfect reconstructing linear phase fractional octave filter bank."
+         "(num_fractions=1, frequency_range=(63, 16000), overlap=1, slope=0)")
+    assert f_obj.sampling_rate == 44100
+
+    # test frequencies
+    _, f_test = pfilt.fractional_octave_frequencies(
+        frequency_range=(63, 16000))
+    npt.assert_allclose(f, f_test)
+
+    # test filtering
+    x = pf.signals.impulse(2**12)
+    y, f = pfilt.reconstructing_fractional_octave_bands(x)
+    assert isinstance(y, Signal)
+    assert y.cshape == (9, )
+    assert y.fft_norm == 'none'
+
+    # test reconstruction (sum has a group delay of half the filter length)
+    reference = pf.signals.impulse(2**12, 2**11)
+    y_sum = y.copy()
+    y_sum.time = np.sum(y_sum.time, 0)
+    npt.assert_allclose(y_sum.time, reference.time, atol=1e-6)
+
+
+def test_reconstructing_fractional_octave_bands_filter_slopes():
+    """Test the shape of the filter slopes for different parameters"""
+    # test different filter slopes against reference
+    x = pf.signals.impulse(2**10)
+
+    for overlap, slope in zip([1, 1, 0], [0, 3, 0]):
+        y, _ = pfilt.reconstructing_fractional_octave_bands(
+            x, frequency_range=(8e3, 16e3), overlap=overlap, slope=slope,
+            n_samples=2**10)
+        reference = np.loadtxt(os.path.join(
+            os.path.dirname(__file__), "references",
+            f"filter.reconstructing_octaves_{overlap}_{slope}.csv"))
+        # restricting rtol was not needed locally. It was added for tests to
+        # pass on travis ci
+        npt.assert_allclose(
+            y.time, np.atleast_2d(reference), rtol=.01, atol=1e-10)
```

Comment:
That's a very small tolerance, is this maybe related to how the data is read? I'm interested where the tolerance is coming from, since it would be a potentially large error in the filter.

Reply:
I puzzled here - saving references in csv files worked witout restrictions in other cases. I thought of different number formats across systems, but that's only a guess.